zero_functional.nim

```diff
@@ -1,171 +1,385 @@
-import strutils, sequtils, macros, options, sets
+import sequtils, macros, options, sets, lists, typetraits, strutils
 
 const iteratorVariableName = "it"
 const accuVariableName = "a"
+const combinationsId = "c"
 const indexVariableName = "idx"
 
 const internalIteratorName = "__" & iteratorVariableName & "__"
 const useInternalAccu = accuVariableName != "result"
 const internalAccuName = if (useInternalAccu): "__" & accuVariableName & "__" else: "result"
-const emptyIdentifier = "__empty__"
+const implicitTypeSuffix = "?" # used when result type is automatically determined
+const listIteratorName = "__itlist__"
+const listIteratorInnerName = "__itlist2__"
 
 type 
+
+  Command {.pure.} = enum
+    ## All available commands.
+    ## 'to' - is a virtual command
+    all, combinations, exists, filter, find, flatten, fold, foreach, index, indexedMap, map, reduce, sub, zip
+
   ExtNimNode = ref object ## Store additional info the current NimNode used in the inline... functions
     node: NimNode     ## the current working node / the current function
     index: int        ## index used for the created iterator - 0 for the first 
     isLastItem: bool  ## true if the current item is the last item in the command chain
     initials: NimNode ## code section before the first iterator where variables can be defined
+    endloop: NimNode  ## code at the end of the for / while loop
```

Comment:
-> endLoop?

Reply:
Sorry, but are you only talking in one-worded sentences?

Reply:
yeah OK - I see - camelCase

Reply:
well; I could reply endsWith? I mean it does not seem to be strictly used - also in the libary.
But I  can change it of course (`ExtNimNode` is also only a private type)

Reply:
OK - done - did a lot of camelCasing :camel: